controller_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package gomaasapi
+
+import (
+	"net/http"
+
+	"github.com/juju/testing"
+	jc "github.com/juju/testing/checkers"
+	"github.com/juju/utils/set"
+	"github.com/juju/version"
+	gc "gopkg.in/check.v1"
+)
+
+type versionSuite struct {
+}
+
+var _ = gc.Suite(&versionSuite{})
+
+func (*versionSuite) TestSupportedVersions(c *gc.C) {
+	for _, apiVersion := range supportedAPIVersions {
+		_, _, err := version.ParseMajorMinor(apiVersion)
+		c.Check(err, jc.ErrorIsNil)
+	}
+}
+
+type controllerSuite struct {
+	testing.CleanupSuite
+	server *SimpleTestServer
```

Comment:
I'd really like to see a shim that allows us to inject JSON responses into the API layer, so we can test without an http server - both here and in juju. To use it in juju it would need to be exposed publicly, which means it would need to not be a gross hack. It can be very basic - just mapping endpoints to a sequence of pre-canned responses. It's how we integrate and expose it that needs thought.